### ISSUE TITLE
Better failure handling

### ISSUE THREAD

**Issue by nicolas-simplex** [Selleo/business_process#1]

**Comment by nicolas-simplex** [comment]
I really like what you have done, the only problem I see is that in order to fail a service needs to return a falsy value, how will the service inform which was the error to the controller then?

**Comment by sockmonk** [comment]
I agree, there needs to be some way to get the error message(s) back from the service, so the controller can give them to the view to be displayed. I generally give my service objects an 'errors' accessor, which is a simple array of string messages meant to be displayed to the user. It should only remain empty if the operation succeeds.

**Comment by stevo** [comment]
Hey Guys :) 

I agree with your concerns and thing that @sockmonk approach is the most lightweight and reasonable. Obviously it is a matter of adding some `attr_reader` and storing errors (or whatever) in instance variable inside the service object. Depending on situation we can raise errors inside service object as well. So nothing hard to implement on our own, however if it is needed more often...

How about providing `error` method - if you call inside service object it and pass some argument, this argument will be stored in an array and will be accessible by `errors` getter outside of service object? I think this will be the simplest solution. What are your thoughts?

**Comment by nicolas-clarolab** [comment]
That was exactly what I had in mind. The `error` ( or  maybe `fail`) method should work something like this:

``` ruby
def initialize
  @success = true
end

def fail(error = nil)
  @error = error
  @success = false
end
```

**Comment by stevo** [comment]
That was now released together with version 1.0.1
